### PR TITLE
Allow change pause container image

### PR DESCRIPTION
**Pull request by DingGGu**
**Issue #1514, if available:**

**Description of changes:**
- Pause container image can be change.
- Changed to not use the ECR API if it is not ECR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Testing Done**
- Add test for pause-container

*[See this guide for recommended testing for PRs.](../doc/CONTRIBUTING.md#testing-changes) Some tests may not apply. Completing tests and providing additional validation steps are not required, but it is recommended and may reduce review time and time to merge.*

### CODE REVIEW

File: files/bootstrap.sh [hunk 1]
```diff
@@ -335,8 +342,11 @@ chown root:root /etc/systemd/system/configure-clocksource.service
...
 systemctl enable --now configure-clocksource
 
-ECR_URI=$(/etc/eks/get-ecr-uri.sh "${AWS_DEFAULT_REGION}" "${AWS_SERVICES_DOMAIN}" "${PAUSE_CONTAINER_ACCOUNT:-}")
-PAUSE_CONTAINER_IMAGE=${PAUSE_CONTAINER_IMAGE:-$ECR_URI/eks/pause}
+PAUSE_CONTAINER_IMAGE="${PAUSE_CONTAINER_IMAGE:-}"
+if [[ -z "${PAUSE_CONTAINER_IMAGE}" ]]; then
+  ECR_URI=$(/etc/eks/get-ecr-uri.sh "${AWS_DEFAULT_REGION}" "${AWS_SERVICES_DOMAIN}" "${PAUSE_CONTAINER_ACCOUNT:-}")
+  PAUSE_CONTAINER_IMAGE="${ECR_URI}/eks/pause"
+fi
 PAUSE_CONTAINER="$PAUSE_CONTAINER_IMAGE:$PAUSE_CONTAINER_VERSION"
```
[the remark this answers is not in the document]
Comment (DingGGu) [reply files/bootstrap.sh]: Okay, Applied that changes.

File: files/bootstrap.sh [hunk 2]
```diff
@@ -34,6 +34,7 @@ function print_help {
   echo "--kubelet-extra-args Extra arguments to add to the kubelet. Useful for adding labels or taints."
   echo "--local-disks Setup instance storage NVMe disks in raid0 or mount the individual disks for use by pods [mount | raid0]"
   echo "--mount-bpf-fs Mount a bpffs at /sys/fs/bpf (default: true)"
+  echo "--pause-container-image The full image to pull the pause container from"
```
Comment (cartermckinnon) [inline files/bootstrap.sh:37]:
```suggestion
  echo "--pause-container-image Image reference for the pause container image. If this flag is used, none of the other --pause-container-* flags take effect."
```

**Review by cartermckinnon — CHANGES_REQUESTED** [review]
Looks fine to me, just suggested a tweak to the flag doc.

**Comment by cartermckinnon** [comment]
@DingGGu you'll need to fix the lint findings, then I'll kick off integration tests 👍

**Comment by DingGGu** [comment]
@cartermckinnon Is there anything else I need to change?

**Pull request closed without merging**